modules/codegen/src/smithy4s/codegen/IR.scala

```diff
@@ -114,15 +118,31 @@ object Field {
 
 }
 
-case class Alt(name: String, realName: String, tpe: Type, hints: List[Hint])
+sealed trait UnionMember {
+  def update(f: Product => Product)(g: Type => Type): UnionMember = this match {
```

Comment:
What's the better name for this function? I can't think right now.. it is kind of a bimap, but not really. Maybe I could rewrite it as a bimap?

Reply:
I thought that, but the fact that both functions are constrained to the same return type as the input type (e.g. Product => Product) made it feel like less of a map.. idk